src/features/patients/pages/PatientRegistration.tsx

```diff
@@ -29,44 +28,39 @@ export const PatientRegistration: FC = () => {
 
   const schemasLoading = patientSchemaIsLoading && patientSurveySchemaIsLoading;
 
-  const onChange = (index: number) => ({ formData }: { formData: any }) => {
-    setFormData((state) => ({ ...state, [index]: formData }));
-  };
-
   const { createPatient, createNameNote } = usePatientApi();
 
-  const onSubmit = () => {
+  const handleClose = () => {
```

Comment:
The changes here are to put the `formData` in the `StepperContext`. Then, we pass the `step` down to children (each 'step', so they can lookup their own data in the context.

The submit then is passed the data to submit, as the context is actually a child of this component and this component has no access to the data.